cmd/contrib/main.go

```diff
@@ -118,17 +118,20 @@ func newPlugin() error {
 					return suggestions
 				},
 			},
-			Validate: func(ans any) error {
+			Validate: func(ans interface{}) error {
 				if str, ok := ans.(string); ok {
-					hasUpper := false
-					for _, char := range str {
-						if unicode.IsUpper(char) {
-							hasUpper = true
+					credNamesContainStr := false
+					for _, name := range credname.ListAll() {
+						if strings.ToLower(name.String()) == strings.ToLower(str) {
+							credNamesContainStr = true
+							break
 						}
 					}
-					if !hasUpper {
-						return errors.New(`credential name must be titlecased, e.g. "Access Key" or "Personal Access Token"`)
+
+					if !credNamesContainStr {
+						return errors.New(`credential name must be one of the suggested, e.g. "Access Key" or "Personal Access Token"`)
```

Comment:
It's a feature that you currently can also use names that are not in the already defined list. We know this list is not exhaustive and there will be new plugins that require additional fields to be defined.

I believe the current behavior is that when you enter a name that's not yet defined, a TODO comment is added for the user to go and define it. That's great imo.

Reply:
I think we can revert your change to the validation while keeping your change to the transform function to achieve the ideal result. What do you think?

Reply:
Aha, I see.
I can offer another improvement for `Validate` function, as the initial one doesn't cover all the cases anyway. For example, such a string will pass `Access token` (`token` in lowercase).

What do you think about changing it to that
```
Validate: func(ans interface{}) error {
				if str, ok := ans.(string); ok {
					words := strings.Split(str, " ")
					for _, word := range words {
						if unicode.IsLower(int32(word[0])) {
							return errors.New(`credential name must be titlecased, e.g. "Access Key" or "Personal Access Token"`)
						}
					}
				}

				return nil
			},
```

With that implementation, we are sure that provided string is really in TitleCase.

Reply:
Is `Validate` called before or after transformation?

I'd expect `Access token` to be automatically changed to `Access Token` because of the transformation function you wrote. If validation happens after transformation, I agree that it should check for casing as well. :+1:

Reply:
Yeah, `Transform` calls after `Validate` and only if validation passes.
I'll update `Validate` with the example I shared above.

Reply:
Since there are some nuances here, I'd love if you could add some unit tests that cover the various combinations (`Access token` that's accepted and transformed to `Access Token`, `Foo bar` that's rejected, `Access Token` that's accepted etc.)